decompression.c

```diff
@@ -106,24 +226,60 @@ void decompress_zwz(const char *file_path, const char *output_dir_path) {
             // printf("Hash of %s: %s and it's original hash value: %s\n", output_file_path, hash, header.hash_value);
         }
     }
+    destroyHashMap(filePathMap);
 }
 
 void do_decompression(const char *source_dir_path, const char *output_dir_path) {
     // Step 1: Iterate through all files in the source directory, only deal with .zwz files
     DIR *dir;
     struct dirent *ent;
-    if ((dir = opendir(source_dir_path)) != NULL) {
-        while ((ent = readdir(dir)) != NULL) {
-            // Check if the file extension is .zwz
-            if (strstr(ent->d_name, ".zwz") != NULL) {
-                char file_path[1024];
-                snprintf(file_path, sizeof(file_path), "%s/%s", source_dir_path, ent->d_name);
-                // Step 2: Process each .zwz file
-                decompress_zwz(file_path, output_dir_path);
+    int readEnd = 0;
+    // char* d_name_copy;
+
+    dir = opendir(source_dir_path);
+    if (dir == NULL) perror("Could not open source directory");
+    else {
+        // multi-threads decompressing .zwz files
+        #pragma omp parallel 
+        {
+            char* d_name_copy;
+
+            // update ent and d_name_copy
+            #pragma omp critial 
```

Comment:
Do you mean `#pragma omp critical` in this context?

Reply:
Sorry for the typo, I changed it to #pragma omp critical and tested it. It worked fine.